config/server.go

```diff
@@ -59,7 +97,7 @@ func serverWsConfigLoader() {
 		ServerMaxConn:     util.MustGetInt("SERVER_WEBSOCKET_MAX_CONN"),
 		ReadBufferSize:    util.MustGetInt("SERVER_WEBSOCKET_READ_BUFFER_SIZE"),
 		WriteBufferSize:   util.MustGetInt("SERVER_WEBSOCKET_WRITE_BUFFER_SIZE"),
-		CheckOrigin:       util.MustGetBool("SERVER_WEBSOCKET_CHECK_ORIGIN"),
+		CheckOrigin:       !util.MustGetBool("SERVER_CORS_ENABLED"),
```

Comment:
if we set SERVER_CORS_ENABLED=true, then CheckOrigin becomes false for websocket upgrader and it should then ideally allow all domains, Isn't that counterintuitive?

Reply:
Not exactly, the control-allow-origin header is set by the middleware, so even if the upgrader allows all domains, the allow-origin header won't be sent as a response header resulting in a cors violation for non whitelisted domains.
The check origin function is an additional overhead, which is technically optional, since we have a cors middleware in place. We can't get rid of it, because the upgrader requires it.

Reply:
ok